src/include/network/abstract_network_command.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include <utility>
+#include "common/macros.h"
+#include "common/managed_pointer.h"
+#include "network/abstract_packet_writer.h"
+#include "network/connection_context.h"
+#include "network/network_defs.h"
+#include "network/network_types.h"
+#include "network/protocol_interpreter.h"
+
+namespace terrier::network {
+
+/**
+ * Interface for the execution of the AbstractNetworkCommands
+ */
+class AbstractNetworkCommand {
```

Comment:
ditto

Reply:
I would say this one should have this base class to prevent having duplicate code of the flush on complete and the `ReadBufferView`